plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerExtendedProperties.java

```diff
@@ -48,6 +49,14 @@ public String getWelcomePageUrl() {
 		IProject project = service.getProject();
 		if (project != null) {
 			List<IRoute> routes = project.getResources(ResourceKind.ROUTE);
+			String url = OpenShiftServerUtils.getRouteURL(server);
+			if(!StringUtils.isEmpty(url)) {
+				for (IRoute route: routes) {
+					if(url.equals(route.getURL())) {
+						return welcomePageUrl = url;
+					}
+				}
+			}
```

Comment:
this block would be an excellent candidate for a method #getRoute(String url) for added code clarity